### PR TITLE
Bugfix unreported crash on load style take 2

### PR DESCRIPTION
**Pull request by elpaso**
Followup #7925 hopefully no crashes anymore, but still I'm unsure about what's the expected behavior here.

@3nids can you have a look please?

### CODE REVIEW

**Review by lbartoletti — APPROVED** [review]

File: src/app/qgsvectorlayerloadstyledialog.cpp [hunk 1]
```diff
@@ -60,6 +60,7 @@ QgsVectorLayerLoadStyleDialog::QgsVectorLayerLoadStyleDialog( QgsVectorLayer *la
 
   // fill style categories
   mModel = new QgsMapLayerStyleCategoriesModel( this );
+  mModel->setShowAllCategories( true );
```
Comment (3nids) [inline src/app/qgsvectorlayerloadstyledialog.cpp]: no, the dialog should not list the "all style" line.

File: src/app/qgsmaplayerstylecategoriesmodel.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ void QgsMapLayerStyleCategoriesModel::setShowAllCategories( bool showAll )
 
 QgsMapLayer::StyleCategory QgsMapLayerStyleCategoriesModel::index2category( const QModelIndex &index ) const
 {
-  return mCategoryList.at( index.row() - ( mShowAllCategories ? 1 : 0 ) );
+  return mCategoryList.at( index.row() + ( mShowAllCategories ? 0 : 1 ) );
 }
```
Comment (3nids) [inline src/app/qgsmaplayerstylecategoriesmodel.cpp]: nope, there is a line shift when all categories are shifted. Will fix this.

**Review by 3nids — CHANGES_REQUESTED** [review]

File: src/app/qgsmaplayerstylecategoriesmodel.cpp (hunk 2 above)
Comment (3nids) [reply src/app/qgsmaplayerstylecategoriesmodel.cpp]: https://github.com/qgis/QGIS/pull/7928

**Comment by elpaso** [comment]
@3nids ok, I'll leave it to you. Can I close this one?

**Comment by 3nids** [comment]
yep, doing it

**Pull request closed without merging**